File: App/app/src/main/java/com/cinema/ticket_booking/data/model/response/PageResponse.java
package com.cinema.ticket_booking.data.model.response;

import com.google.gson.annotations.SerializedName;
import java.util.List;

public class PageResponse<T> {
    @SerializedName("content")
    private List<T> content;
    @SerializedName("page")
    private int page;
    @SerializedName("size")
    private int size;
    @SerializedName("totalElements")
    private long totalElements;
    @SerializedName("totalPages")
    private int totalPages;
    @SerializedName("last")
    private boolean last;

    public List<T> getContent() {
        return content;
    }

    public int getTotalPages() {
        return totalPages;
    }

    public boolean isLast() {
        return last;
    }

    public long getTotalElements() {
        return totalElements;
    }

    public void setContent(List<T> content) {
        this.content = content;
    }

    public void setLast(boolean last) {
        this.last = last;
    }

    public void setTotalElements(long totalElements) {
        this.totalElements = totalElements;
    }

    public void setPage(int page) {
        this.page = page;
    }

    public void setSize(int size) {
        this.size = size;
    }

    public void setTotalPages(int totalPages) {
        this.totalPages = totalPages;
    }
}

File: App/app/src/main/java/com/cinema/ticket_booking/data/model/response/PageResponse.kt
package com.cinema.ticket_booking.data.model.response

import com.google.gson.annotations.SerializedName

data class PageResponse<T> @JvmOverloads constructor(
    @SerializedName("content")
    var content: List<T>? = null,

    @SerializedName("page")
    var page: Int = 0,

    @SerializedName("size")
    var size: Int = 0,

    @SerializedName("totalElements")
    var totalElements: Long = 0L,

    @SerializedName("totalPages")
    var totalPages: Int = 0,

    @SerializedName("last")
    @get:JvmName("isLast")
    var last: Boolean = false
)
